gapis/service/types/types.proto

```diff
@@ -0,0 +1,89 @@
+// Copyright (C) 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+import "core/data/pod/pod.proto";
+
+package types;
+option java_package = "com.google.gapid.proto.service.types";
+option java_outer_classname = "TypeInfo";
+option go_package = "github.com/google/gapid/gapis/service/types";
+
+message Type {
+  // type_id is used to describe cyclic / 'diamond' types. The identifiers are
```

Comment:
What is a cyclic/diamond type?

Reply:
This command is no longer valid for this proto. But in general a diamond type is 
```
Foo {
    Bar b1
    Baz b2
}

Bar {
   Foo2 f
}
Baz {
   Foo2 f
}
```
Where the definition for Foo2 would get hit 2 times at the same time if we just serialized straight-line. Meaning we would transmit lots of redundant type data.

Also cyclic types are things like
```
LinkedListNode {
   LinkedListNode* node
}
```
Where we have to break the cycle in order to serialize the possible types.

The types.proto does not work like that however. It does not actually encode the full type-tree, it just returns type_ids for child types, meaning you make separate requests for different parts of the tree. This was a holdover from our original box.go.